moto/timestreamwrite/responses.py

```diff
@@ -85,7 +86,12 @@ def describe_table(self) -> str:
 
     def list_tables(self) -> str:
         database_name = self._get_param("DatabaseName")
-        tables = self.timestreamwrite_backend.list_tables(database_name)
+        if database_name is None:
+            tables: Iterable[TimestreamTable] = []
+            for db in self.timestreamwrite_backend.list_databases():
+                tables = list(tables) + list(db.list_tables())
+        else:
+            tables = self.timestreamwrite_backend.list_tables(database_name)
         return json.dumps(dict(Tables=[t.description() for t in tables]))
```

Comment:
Is there a reason you put this logic here and not in `timestreamwrite_backend.list_tables`?

Reply:
Good point, move code to backend. Thanks.